tests/switchless_worksleep/host/host.cpp

```diff
@@ -0,0 +1,214 @@
+// Copyright (c) Open Enclave SDK contributors.
+// Licensed under the MIT License.
+
+#include <inttypes.h>
+#include <limits.h>
+#include <openenclave/host.h>
+#include <openenclave/internal/error.h>
+#include <openenclave/internal/tests.h>
+#include <openenclave/internal/thread.h>
+#include <chrono>
+#include <cstdio>
+#include <cstdlib>
+#include <thread>
+#include <vector>
+#include "../../../host/hostthread.h"
+#include "../../../host/strings.h"
+#include "switchless_worksleep_u.h"
+using namespace std;
+
+#if defined(__linux__)
+
+#include <sys/sysinfo.h>
+
+#elif defined(_WIN32)
+
+#include <Windows.h>
+#include <intrin.h>
+
+uint32_t get_nprocs(void)
+{
+    SYSTEM_INFO system_info;
+    ZeroMemory(&system_info, sizeof(system_info));
+
+    GetSystemInfo(&system_info);
+    return static_cast<size_t>(system_info.dwNumberOfProcessors);
```

Comment:
You can get this in a cross platform way in C++: `std::thread::hardware_concurrency();`

Reply:
Great! Thanks